### ISSUE TITLE
some tests not compatible to OpenMP

### ISSUE THREAD

**Issue by the-hampel** [TRIQS/nda#12]
Wenn using `-fopenmp` as CXXFLAG explicitly some tests do not compile. Error output for one of them:
```

In file included from /mnt/home/ahampel/Dropbox/git/triqs3/nda.src/c++/nda/declarations.hpp:20:                                                                                                                                                                                                                                                                                              
/mnt/home/ahampel/Dropbox/git/triqs3/nda.src/c++/nda/storage/policies.hpp:50:5: error: static_assert failed due to requirement 'false && always_true<nda::allocators::segregator<800, nda::allocators::multiple_bucket<800>, nda::allocators::mallocator> >' "Custom Allocators are not available in OpenMP"                                                                                 
    static_assert(false and always_true<Allocator>, "Custom Allocators are not available in OpenMP");                                                                                                                                                                                                                                                                                        
    ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                           
/mnt/home/ahampel/Dropbox/git/triqs3/nda.src/c++/nda/basic_array.hpp:78:32: note: in instantiation of template class 'nda::heap_custom_alloc<nda::allocators::segregator<800, nda::allocators::multiple_bucket<800>, nda::allocators::mallocator> >' requested here                                                                                                                          
    using storage_t = typename ContainerPolicy::template handle<ValueType, layout_t::ce_size()>;                                                                                                                                                                                                                                                                                             
                               ^                                                                                                                                                                                                                                                                                                                                                             
/mnt/home/ahampel/Dropbox/git/triqs3/nda.src/test/c++/nda_custom_alloc.cpp:24:77: note: in instantiation of template class 'nda::basic_array<long, 2, nda::C_layout, 'A', nda::heap_custom_alloc<nda::allocators::segregator<800, nda::allocators::multiple_bucket<800>, nda::allocators::mallocator> > >' requested here                                                                    
  nda::basic_array<long, 2, C_layout, 'A', nda::heap_custom_alloc<alloc_t>> A(3, 3);                                                                                                                                                                                                                                                                                                         
                                                                            ^                                                                                                                                                                                                                                                                                                                
In file included from /mnt/home/ahampel/Dropbox/git/triqs3/nda.src/test/c++/nda_custom_alloc.cpp:15:                                                                                                                                                                                                                                                                                         
In file included from /mnt/home/ahampel/Dropbox/git/triqs3/nda.src/test/c++/./test_common.hpp:25:                                                                                                                                                                                                                                                                                            
In file included from /mnt/home/ahampel/Dropbox/git/triqs3/nda.src/c++/nda/nda.hpp:28:                                                                                                                                                                                                                                                                                                       
/mnt/home/ahampel/Dropbox/git/triqs3/nda.src/c++/nda/basic_array.hpp:78:58: error: no member named 'handle' in 'nda::heap_custom_alloc<nda::allocators::segregator<800, nda::allocators::multiple_bucket<800>, nda::allocators::mallocator> >'                                                                                                                                               
    using storage_t = typename ContainerPolicy::template handle<ValueType, layout_t::ce_size()>;                                                                                                                                                                                                                                                                                             
                               ~~~~~~~~~~~~~~~~~         ^                                                                                                                                                                                                                                                                                                                                   
/mnt/home/ahampel/Dropbox/git/triqs3/nda.src/test/c++/nda_custom_alloc.cpp:24:77: note: in instantiation of template class 'nda::basic_array<long, 2, nda::C_layout, 'A', nda::heap_custom_alloc<nda::allocators::segregator<800, nda::allocators::multiple_bucket<800>, nda::allocators::mallocator> > >' requested here                                                                    
  nda::basic_array<long, 2, C_layout, 'A', nda::heap_custom_alloc<alloc_t>> A(3, 3);                                                                                                                                                                                                                                                                                                         
                                                                            ^                 
                                                     
```
We either need to compile always without openmp, fix the test, or not compile the test.

**Comment by parcollet** [comment]
We should not compile the test in openmp mode...
Remove the test for Cmake if openMP flags is on

**Comment by Wentzell** [comment]
This was fixed in ad8a9c4bd8051896f9dda45a68c55ae98050c7a7
